### PR TITLE
OboeTester: catch null from getChannelMapping()

### LINKED ISSUE #1715: OboeTester: DeviceReport crashing on null mapping
micInfo.getChannelMapping()
is supposed to return an "Empty list when the MicrophoneInfo is returned by AudioManager.getMicrophones()."
See [b/270537789](http://b/270537789) | P4 | [Oboe Tester] : Oboe tester app crashed when click on DEVICE REPORT

Detect the null and show a Toast.

### PR DESCRIPTION
**Pull request by philburk**
Fixes #1715

### CODE REVIEW

File: apps/OboeTester/app/src/main/java/com/mobileer/oboetester/DeviceReportActivity.java [hunk 1]
```diff
@@ -127,8 +129,12 @@ public String reportAllMicrophones() {
                     report.append(micItem);
                 }
             } catch (IOException e) {
-                e.printStackTrace();
+                Log.e(TestAudioActivity.TAG, "Caught ", e);
```
Comment (robertwu1) [inline apps/OboeTester/app/src/main/java/com/mobileer/oboetester/DeviceReportActivity.java:132]: Should we show error toast for IOExceptions?

File: apps/OboeTester/app/src/main/java/com/mobileer/oboetester/MicrophoneInfoConverter.java [hunk 2]
```diff
@@ -60,6 +60,9 @@ public static String reportMicrophoneInfo(MicrophoneInfo micInfo) {
         sb.append("\nType       : " + micInfo.getType());
 
         List<Pair<Integer, Integer>> mapping = micInfo.getChannelMapping();
+        if (mapping == null) {
```
Comment (robertwu1) [inline apps/OboeTester/app/src/main/java/com/mobileer/oboetester/MicrophoneInfoConverter.java:63]: Are we sure null is an error? What about if this is empty?

The documentation for the function seems a bit weird https://developer.android.com/reference/android/media/MicrophoneInfo#getChannelMapping()

**Review by robertwu1 — CHANGES_REQUESTED** [review]

File: apps/OboeTester/app/src/main/java/com/mobileer/oboetester/MicrophoneInfoConverter.java (hunk 2 above)
Comment (philburk) [reply apps/OboeTester/app/src/main/java/com/mobileer/oboetester/MicrophoneInfoConverter.java:63]: Jiabin says yes, it is a regression.

File: apps/OboeTester/app/src/main/java/com/mobileer/oboetester/DeviceReportActivity.java (hunk 1 above)
Comment (philburk) [reply apps/OboeTester/app/src/main/java/com/mobileer/oboetester/DeviceReportActivity.java:132]: We did not do it before and I assume there was a reason, which I don't recall.

**Pull request merged**